Nos/Controller/SearchController.swift

```diff
@@ -101,8 +146,28 @@ class SearchController: ObservableObject {
         }
     }
     
-    func clear() {
-        query = ""
-        authorSuggestions = []
+    func submitSearch() { // rename to seeIfThisIsSomeSortOfIdentifier (or maybe put this all into .map)
+        if query.contains("@") {
+            Task(priority: .userInitiated) {
+                if let publicKeyHex =
+                    await relayService.retrievePublicKeyFromUsername(query.lowercased()) {
+                    Task { @MainActor in
+                        if let author = author(fromPublicKey: publicKeyHex) {
+                            router.push(author)
+                        }
+                    }
+                }
+            }
+        } else {
+            if let author = author(fromPublicKey: query) {
```

Comment:
There is a problem here, that could be outside the scope of this PR because its an issue we already had before. The implementation of author(fromPublicKey:), after checking if it is a valid npub key, just defaults to an hex string with npub prefix. This makes it impossible to search for an hex-encoded note identifier (for instance: 3a20b9cc890c8cb11918eec679265b3d97a28f781fbb1bd31d46ccaceeafab19) because it parses it as a npub and just opens a profile (that doesn't exist).

Reply:
I think this is a good thing to open a ticket for. @joshuatbrown I think what we discussed was the ability to open a profile by its hex ID instead of npub being a nice-to-have debugging feature. Looking up notes by their `note123`-style ID (aka Bech32 ID) is a user-facing feature.

Reply:
@martindsq would you open the ticket for this? I'll get this PR merged.